packages/core/src/components/spinner/spinner.tsx

```diff
@@ -19,12 +19,18 @@ const SPINNER_TRACK = "M 50,50 m 0,-44.5 a 44.5,44.5 0 1 1 0,89 a 44.5,44.5 0 1
 // this value is the result of `<path d={SPINNER_TRACK} />.getTotalLength()` and works in all browsers:
 const PATH_LENGTH = 280;
 
-export interface ISpinnerProps extends IProps, IIntentProps {
-    /** Whether this spinner should use large styles. */
-    large?: boolean;
+const MIN_SIZE = 10;
+const STROKE_WIDTH = 4;
+const MIN_STROKE_WIDTH = 16;
 
-    /** Whether this spinner should use small styles. */
-    small?: boolean;
+export interface ISpinnerProps extends IProps, IIntentProps {
+    /**
+     * Width and height of the spinner in pixels. The size cannot be less than
+     * 10px. Constants are available for common sizes: `Spinner.SIZE_SMALL`,
+     * `Spinner.SIZE_STANDARD`, `Spinner.SIZE_LARGE`.
+     * @default Spinner.SIZE_STANDARD = 50
+     */
+    size?: number;
```

Comment:
`iconSize` only allows passing in `16` or `20`. Do we really want to support anything here?

Reply:
nope, you can pass any `number`: http://blueprintjs.com/docs/v2/#core/components/icon

but since icons are hinted at 16 and 20, you shouldn't use any other size unless you know exactly what you're doing. not true for spinners, they look good at any size